File: public/js/game/commands/addRoomUserCommand.js
function AddRoomUserCommand(roomId){
    Command.call(this, AddRoomUserCommand.ID);
    this.roomId = roomId
    this.data = true;
    this.execute = function(){
        var user = new GameUser(this.data.user);
        if(!Model.room.containsUser(user)){
            Model.room.addUser(user);
            Model.dispatch(Events.ROOM_USERS_CHANGE, user)
            Model.dispatch(Events.ROOM_USER_ADD, user)
            trace("User uid=" + user.uid + " added to room #" + Model.room.roomId);
        }
    }
}
AddRoomUserCommand.ID = 'adduser'
Command.add(AddRoomUserCommand.ID, AddRoomUserCommand)

File: src/main/java/com/hellsepontus/commands/AddRoomUserCommand.java
package com.hellsepontus.commands;

import com.hellsepontus.model.GameUser;
import com.hellsepontus.model.Room;

import java.util.Collections;

public class AddRoomUserCommand extends Command {

    public static final String ID = "adduser";
    public GameUser user;
    public Room room;
    
    public AddRoomUserCommand(Room room, GameUser user) {
        super(ID);
        this.user = user;
        this.room = room;
    }

    @Override
    public void execute() {
        sendToAll(user.getRoom(), ID, Collections.singletonMap("user", (Object) user));
    }

    @Override
    public GameUser getUser() {
        return user;
    }
}
